src/SocialPostManager.php

```diff
@@ -0,0 +1,260 @@
+  protected $key;
+
+  /**
+   * The implementer plugin id.
+   *
+   * @var string
+   */
+  protected $pluginId;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   Used for logging errors.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Used for loading and creating Drupal user objects.
+   * @param \Drupal\Core\Entity\Query\QueryFactory $entityQuery
+   *   Used to get entity query object for this entity type.
+   * @param \Drupal\Core\Session\AccountProxy $current_user
+   *   Used to get current active user.
+   */
+  public function __construct(LoggerChannelFactoryInterface $logger_factory, EntityTypeManagerInterface $entity_type_manager, QueryFactory $entityQuery, AccountProxy $current_user) {
+    $this->loggerFactory = $logger_factory;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityQuery = $entityQuery;
+    $this->currentUser = $current_user;
+
+    $this->key = $this->getSalt();
+    $this->setPluginId('social_post');
+  }
+
+  /**
+   * Sets the implementer plugin id.
+   *
+   * This value is used to generate customized logs, drupal messages, and event
+   * dispatchers.
+   *
+   * @param string $plugin_id
+   *   The plugin id.
+   */
+  public function setPluginId($plugin_id) {
+    $this->pluginId = $plugin_id;
+  }
+
+  /**
+   * Gets the implementer plugin id.
+   *
+   * This value is used to generate customized logs, drupal messages, and events
+   * dispatchers.
+   *
+   * @return string
+   *   The plugin id.
+   */
+  public function getPluginId() {
+    return $this->pluginId;
+  }
+
+  /**
+   * Add user record in Social Post Entity.
+   *
+   * @param string $pluginId
+   *   Type of social network.
+   * @param string $provider_user_id
+   *   Unique Social ID returned by social network.
+   * @param string $token
+   *   Token to be used for autoposting.
+   *
+   * @return True
+   *   if User record was created or
+   *   False otherwise
+   */
+  public function addRecord($pluginId, $provider_user_id, $token) {
+    // Get User ID of logged in user.
+    $user_id = $this->currentUser->id();
+
+    // Encode token into json format.
+    $json_token = json_encode($token);
+
+    // Add user record.
+    $values = [
+      'user_id' => $user_id,
+      'plugin_id' => $pluginId,
+      'provider_user_id' => $provider_user_id,
+      'token' => $this->encryptToken($json_token),
+    ];
+
+    $user_info = $this->entityTypeManager->getStorage('social_post')->create($values);
+    if ($user_info) {
+      return TRUE;
+    }
+    else {
+      return FALSE;
+    }
+
+  }
+
+  /**
+   * Update token of a praticular record.
+   *
+   * @param string $pluginId
+   *   Type of social network.
+   * @param string $provider_user_id
+   *   Unique Social ID returned by social network.
+   * @param string $token
+   *   Token provided by social_network.
+   *
+   * @return True if updated.
```

Comment:
1) Description of return value should be in the next line.
2) I don't see any return statement in the method
3) Shouldn't the method return FALSE if record could not be updated?